api/server/httprest/server.go

```diff
@@ -99,3 +99,7 @@ func (g *Server) Stop() error {
 	}
 	return nil
 }
+
+func (g *Server) RegisterMiddlewares(mw middleware.Middleware) {
```

Comment:
this function doesn't actually register the middleware.. it just appends the middleware to the array, registering it happens when you are assigning it in the server's handler...